### PR TITLE
Add Python Typing

### PR DESCRIPTION
**Pull request by sobolevn**
https://github.com/typeddjango/awesome-python-typing

Collection of awesome Python types, stubs, plugins, and tools to work with them.

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- [x] **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- [x] **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
       Links: [#1636](https://github.com/sindresorhus/awesome/pull/1636) and [#1638](https://github.com/sindresorhus/awesome/pull/1638)

- [x] You have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- [x] This pull request has a title in the format `Add Name of List`.
	- ✅ `Add Python Typing`
- [x] Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [Typing](…) - Types and tools to work with them for Python.`
- [x] Your entry should be added at the bottom of the appropriate category.
- [x] The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- [x] **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- [x] Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it.
- [x] Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new). Link to our CI: https://travis-ci.com/typeddjango/awesome-python-typing/builds/133380966
- [x] **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Collection of awesome Python types, stubs, plugins, and tools to work with them.`
- [x] The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-python-typing`
- [x] The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Python Typing`
- [x] Non-generated Markdown file in a GitHub repo.
- [x] The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- [x] Not a duplicate. Please search for existing submissions.
- [x] Only has awesome items. Awesome lists are curations of the best, not everything.
- [x] Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- [ ] Includes a project logo/illustration whenever possible.
	- [ ] Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- [ ] The image should link to the project website or any relevant website.
	- [ ] **The image should be high-DPI.** Set it to maximum half the width of the original image.
- [x] Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- [x] Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
	- [x] Should be placed on the right side of the readme heading.
		- [ ] Can be placed centered if the list has a centered graphics header.
	- [x] Should link back to this list.
- [x] Has a Table of Contents section.
	- [x] Should be named `Contents`, not `Table of Contents`.
	- [x] Should be the first section in the list.
	- [x] Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
- [x] Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
	- [x] That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**
	- [x] [WTFPL](http://www.wtfpl.net) and [Unlicense](http://unlicense.org) are not acceptable licenses.
	- [ ] **If you use a license badge, it should be SVG, not PNG.**
	- [x] To verify that you've read all the guidelines, please add comment with just the word `unicorn`.
- [x] Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
	- [x] The file should be named `contributing.md`. Casing is up to you.
- [x] Has consistent formatting and proper spelling/grammar.
	- [x] The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- [x] The description starts with an uppercase character and ends with a period.
	- [x] Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- [x] Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- [x] Doesn't include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

### CODE REVIEW

**Comment by sobolevn** [comment]
unicorn

File: readme.md [hunk 1]
```diff
@@ -146,6 +146,7 @@
 	- [Scientific Audio](https://github.com/faroit/awesome-python-scientific-audio#readme) - Scientific research in audio/music.
 	- [CircuitPython](https://github.com/adafruit/awesome-circuitpython#readme) - A version of Python for microcontrollers.
 	- [Data Science](https://github.com/krzjoa/awesome-python-data-science#readme) - Data analysis and machine learning.
+	- [Typing](https://github.com/typeddjango/awesome-python-typing) - Types and tools to work with them for Python.
```
Comment (sindresorhus) [inline readme.md]:
```suggestion
	- [Typing](https://github.com/typeddjango/awesome-python-typing#readme) - Types and tools to work with them for Python.
```

And the description should be about what "typing" means for Python, not what the list contains.

**Comment by sindresorhus** [comment]
>  Includes a succinct description of the project/theme at the top of the readme. (Example)
✅ Collection of awesome Python types, stubs, plugins, and tools to work with them.

File: readme.md (hunk 1 above)
Comment (sobolevn) [reply readme.md]: Ok, changed the wording. Thanks!

**Comment by sindresorhus** [comment]
You missed something in https://github.com/sindresorhus/awesome/pull/1639#discussion_r339416284.

**Comment by sobolevn** [comment]
Added `#readme` target. Thanks!

**Comment by sindresorhus** [comment]
>  Includes a succinct description of the project/theme at the top of the readme. (Example)

The top of your readme should explain what the optional typing in Python is, and link to some more reading about it.

**Comment by sindresorhus** [comment]
Tweet: https://twitter.com/awesome__re/status/1195609483063349248

**Comment by sobolevn** [comment]
Thanks!

**Pull request merged**